services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/serialization/TransferManagerJsonMarshaller.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.transfer.s3.internal.serialization;
+
+import static software.amazon.awssdk.transfer.s3.internal.serialization.TransferManagerMarshallingUtils.getMarshaller;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.core.util.SdkAutoConstructList;
+import software.amazon.awssdk.core.util.SdkAutoConstructMap;
+import software.amazon.awssdk.protocols.core.Marshaller;
+import software.amazon.awssdk.protocols.jsoncore.JsonWriter;
+import software.amazon.awssdk.utils.DateUtils;
+
+/**
+ * Interface to marshall data according to the JSON protocol specification.
+ *
+ * @param <T> Type to marshall.
+ */
+@FunctionalInterface
+@SdkInternalApi
+public interface TransferManagerJsonMarshaller<T> extends Marshaller<T> {
+
+    TransferManagerJsonMarshaller<String> STRING = (val, jsonGenerator) -> jsonGenerator.writeValue(val);
+    TransferManagerJsonMarshaller<Short> SHORT = (val, jsonGenerator) -> jsonGenerator.writeValue(val);
+    TransferManagerJsonMarshaller<Integer> INTEGER = (val, jsonGenerator) -> jsonGenerator.writeValue(val);
+    TransferManagerJsonMarshaller<Long> LONG = (val, jsonGenerator) -> jsonGenerator.writeValue(val);
+    TransferManagerJsonMarshaller<Float> FLOAT = (val, jsonGenerator) -> jsonGenerator.writeValue(val);
+    TransferManagerJsonMarshaller<Double> DOUBLE = (val, jsonGenerator) -> jsonGenerator.writeValue(val);
+    TransferManagerJsonMarshaller<BigDecimal> BIG_DECIMAL = (val, jsonGenerator) -> jsonGenerator.writeValue(val);
+    TransferManagerJsonMarshaller<Boolean> BOOLEAN = (val, jsonGenerator) -> jsonGenerator.writeValue(val);
+    TransferManagerJsonMarshaller<Instant> INSTANT =
+        (val, jsonGenerator) -> jsonGenerator.writeNumber(DateUtils.formatUnixTimestampInstant(val));
+    TransferManagerJsonMarshaller<SdkBytes> SDK_BYTES = (val, jsonGenerator) -> jsonGenerator.writeValue(val.asByteBuffer());
+
```

Comment:
Can we just use `writeValue(val)`?

Reply:
I added a convenience method to the jsonwriter:
~~~
    public JsonWriter writeValue(SdkBytes value) {
        writeValue(value.asByteBuffer());
        return this;
    }

Reply:
Actually adding this method caused type erasure conflicts, removed it.

Reply:
Sorry, I commented on the wrong line. I meant line 49 `TransferManagerJsonMarshaller<Instant> INSTANT = ....`